OData/src/System.Web.OData/OData/Routing/Conventions/AttributeRoutingConvention.cs

```diff
@@ -205,15 +205,17 @@ public string SelectController(ODataPath odataPath, HttpRequestMessage request)
                 {
                     IHttpActionSelector actionSelector = controller.Configuration.Services.GetActionSelector();
                     ILookup<string, HttpActionDescriptor> actionMapping = actionSelector.GetActionMapping(controller);
-                    IEnumerable<HttpActionDescriptor> actions = actionMapping.SelectMany(a => a);
-                    string prefix = GetODataRoutePrefix(controller);
-
-                    foreach (HttpActionDescriptor action in actions)
+                    HttpActionDescriptor[] actions = actionMapping.SelectMany(a => a).ToArray();
+ 
+                    foreach (string prefix in AttributeRoutingConvention.GetODataRoutePrefixes(controller))
                     {
-                        IEnumerable<ODataPathTemplate> pathTemplates = GetODataPathTemplates(prefix, action);
-                        foreach (ODataPathTemplate pathTemplate in pathTemplates)
+                        foreach (HttpActionDescriptor action in actions)
                         {
-                            attributeMappings.Add(pathTemplate, action);
+                            IEnumerable<ODataPathTemplate> pathTemplates = GetODataPathTemplates(prefix, action);
+                            foreach (ODataPathTemplate pathTemplate in pathTemplates)
+                            {
+                                attributeMappings.Add(pathTemplate, action);
```

Comment:
The existing code already seemed to handle the scenario by virtue of how the templates are stored in the dictionary. So:
[ODataRoute("entityset({key})")]
[ODataRoute("entityset({key})")]

Ended up collapsing to a single entry since they are duplicates.

Because of that I didn't worry about doing anything special here. We *could* potentially detect duplicates and throw an error but semantically it didn't seem worth while - there's no ambiguity introduced by having it declare twice, if anything it just pollutes the source.